### PR TITLE
es_type_overrides should error when receiving non-existing entries

### LINKED ISSUE #280: es_type_overrides should error when receiving non-existing entries

### PR DESCRIPTION
**Pull request by V1NAY8**
Closes #280 

@sethmlarson Please review 😄

### CODE REVIEW

**Comment by sethmlarson** [comment]
Jenkins test this please

**Comment by V1NAY8** [comment]
Builds are good :)

File: eland/field_mappings.py [hunk 1]
```diff
@@ -537,7 +535,17 @@ def _generate_es_mappings(
         }
         """
 
-        mapping_props = {}
+        mapping_props: Dict[str, Any] = {}
+        es_dtype: str
```
Comment (sethmlarson) [inline eland/field_mappings.py:539]: Is this type hint necessary? If it is let's move it to the top of the function with whitespace before the `mapping_props = {}` so we don't confuse it for an initialization.

File: eland/field_mappings.py [hunk 2]
```diff
@@ -479,7 +477,7 @@ def _pd_dtype_to_es_dtype(pd_dtype):
         category NA NA Finite list of text values
         ```
         """
-        es_dtype = None
+        es_dtype: str = None
```
Comment (sethmlarson) [inline eland/field_mappings.py:480]: This function can return `None` so should be `Optional[str]` both here and return type?

File: eland/field_mappings.py [hunk 3]
```diff
@@ -537,7 +535,17 @@ def _generate_es_mappings(
...
-        mapping_props = {}
+        mapping_props: Dict[str, Any] = {}
+        es_dtype: str
+        if es_type_overrides is not None:
+            non_existing_columns: List[str] = [
+                key for key in es_type_overrides.keys() if key not in dataframe.columns
+            ]
+            if non_existing_columns:
+                raise KeyError(
+                    f"{repr(non_existing_columns)[1:-1]} column/s not in given dataframe"
```
Comment (sethmlarson) [inline eland/field_mappings.py:546]: nit: `column(s)` instead of `column/s`

**Review by sethmlarson — CHANGES_REQUESTED** [review]
Few small comments then looks good!

File: eland/field_mappings.py (hunk 1 above)
Comment (V1NAY8) [reply eland/field_mappings.py]: Just added, Anyway will move it as suggested

File: eland/field_mappings.py (hunk 2 above)
Comment (V1NAY8) [reply eland/field_mappings.py]: Yep, Missed in plain sight 👍

**Comment by V1NAY8** [comment]
CI workflows are failing because, we are using elasticsearch 
Hence we need to add `python3.8 -m pip install nox elasticsearch` in yml

You want me to add it?

**Comment by sethmlarson** [comment]
Instead of installing `elasticsearch` can you move the `import elasticsearch` from the global scope to inside the `try-except` for `es_active`?

**Comment by V1NAY8** [comment]
Hmm. Now pandoc is not installed 🤯

**Comment by sethmlarson** [comment]
ah, likely need a `apt-get install pandoc` within the github action?

**Comment by sethmlarson** [comment]
@V1NAY8 I pushed to your branch, recommend you do a pull.

**Comment by V1NAY8** [comment]
Yes, pulled it, Now the CI are successful .

**Review by sethmlarson — APPROVED** [review]
Woo, thank you for this! :tada: :partying_face:

**Pull request merged**